auth added

Meelenych opened the pull request.

Meelenych committed:
auth added

Meelenych committed:
old file deleted

Meelenych committed:
.env fixed + middleware added

Meelenych committed:
logout and pagination added

Meelenych committed:
logs deleted

KirillKulakow commented on routes/api/contacts-routes.js:
```diff
@@ -0,0 +1,143 @@
+	try {
+		const { page = 1, limit = 10 } = req.query;
+		const { _id } = req.user;
+		const skip = (page - 1) * limit;
```
перед математическими операциями рекомендую проверять входящие параметры query
например через метод isNaN

Meelenych replied:
Пытался сделать. Потом начало выбивать ошибку. Не могу понять почему.
 {
    "message": "Cannot do inclusion on field limit in exclusion projection"
}
Закоментировал то, что сделал.

KirillKulakow commented on routes/api/contacts-routes.js:
```diff
@@ -0,0 +1,143 @@
+			throw createError(400, "invalid ID");
+		}
+		const result = await Contact.findByIdAndDelete(contactId);
+		if (!result || !contactId) {
```
смысл проверки !contactId? если он у тебя будет undefined - ошибка будет на строке 78

Meelenych replied:
Убрал.

KirillKulakow commented on routes/api/contacts-routes.js:
```diff
@@ -0,0 +1,143 @@
+			new: true,
+		});
+
+		if (req.body === null) {
```
зачем тебе эта проверка, если проверка выполняется на строке 93

Meelenych replied:
Убрал.

KirillKulakow commented on routes/api/contacts-routes.js:
```diff
@@ -0,0 +1,143 @@
+const express = require("express");
```
исходя из твоего приложения - любой зарегистрированный юзер может удалить/изменить/посмотреть любой контакт в бд
сделай поиск не только по ид контакта но и с учетом ид собственника контакта

Meelenych replied:
Поправил. Был неправильный метод поиска в контроллере.

KirillKulakow replied:
проблема не исправлена
любой зарегистрированный юзер может делать операции с любым контактом (не важно его этот контакт или нет)

Meelenych replied:
Добавил owner ID

KirillKulakow reviewed (changes requested):
Need fix

Meelenych committed:
adds page and limit conversion to Number

Meelenych committed:
removes excessive checks

Meelenych committed:
token fix

Meelenych committed:
Pagination removed

Cased Error {
    "message": "Cannot do inclusion on field limit in exclusion projection"
}

KirillKulakow reviewed (changes requested):
Need fix

Meelenych committed:
owner ID added

Meelenych committed:
pagination disabled
